### ISSUE TITLE
Disable dark shadows in UMD mode

### ISSUE THREAD

**Issue by lianee** [quasarframework/quasar#15144]
### What happened?

The new 'dark' shadows look more like a lighting behind elements than a real shadow and is a strong visual change. While I understand and appreciate a special kind of shadow in dark mode, I'd like to adapt it to my personal taste.

### What did you expect to happen?

A way to revert it

### Reproduction URL

https://jsfiddle.net/q2drzm45/4/

### How to reproduce?

upgrade to 2.11

### Flavour

UMD

### Areas

Components (quasar)

### Additional context

I can easily revert it in cli app like that:
```scss
$separator-dark-color: rgba(0, 0, 0, 0.12);
$dark-shadow-color: #000;
$layout-shadow-dark: 0 0 10px 2px rgba(0, 0, 0, 0.2), 0 0px 10px rgba(0, 0, 0, 0.24);
```
Unfortunately, I haven't been able to do the same in UMD mode with the config object. Dark shadows are not using css vars.

**Comment by agroebe** [comment]
My $0.02 would be that the feature should just be reverted entirely. I think these shadows are desirable on light mode backgrounds for more contrast, and I'm glad that this was the default. But I don't think they belong in dark mode and they look horrible in my opinion. I had to go through and manually add `flat` to all affected components, which removed the desirable contrast in light mode that I talked about.

If not, might be good to have an "officially documented" way to disable this feature, so the community doesn't need to monkeypatch a solution together to make their app look the same it did previously.

Either way, +1 on the opinion that the visual change is way too strong to be a "default" behavior.

**Comment by exninja** [comment]
@agroebe 
The workaround I put into my apps so the new shadow is ignored in dark mode only.

`:flat="$q.dark.isActive"`

**Comment by piperone** [comment]
I'm glad to find others of this opinion. I just updated one of my projects to the latest version and immediately noticed the white'ish gradient box-shadow behind dropdowns, dialogs, etc. Given this box-shadow change *and* the new docs-site, it looks like someone on the Quasar-team must be really fond of 2010-vintage style box-shadows.

I echo @agroebe's sentiment; the white box-shadow in dark mode should be reverted.

**Comment by RobsonMi** [comment]
I agree these new shadows look terrible and should be reverted! Right now instead of providing updates to users I must search a workaround to remove them completely. What design decision is that to add bright, cheap looking, 2000's glow to dark theme?!

**Comment by lianee** [comment]
I made a codepen to change the color of the shadows using CSS variables: [https://codepen.io/liane_/pen/JjaQZVw](https://codepen.io/liane_/pen/JjaQZVw)
Unfortunately, due to the limitations of CSS, the format used for the color is awkward (0,0,0), that's why I didn't propose a Pull Request.
I also found that the shadows in the dark version, if they are black (as I prefer), lack a little presence, so I added an optional multiplier (--q-dark-shadow-opacity). Personally, I find that these Dark shadows look better with this multiplier set to 2

**Comment by Flamenco** [comment]
To remove box shadows in dark mode, I updated the CSS for components instead of changing the attribute to `flat`.

For example
```css
.q-card.q-dark {
  border: 1px solid #fff3;
  box-shadow: none;
}
```

This, along with changing the default heading font sizes to sane values, is in every Quasar project I manage.

**Comment by eliaspanagiotidis** [comment]
Neither Material 2 nor Material 3 accommodate light shadows in dark mode. Instead, they both advocate for the retention of darker shadows, particularly when an elevated element is superimposed on other elements. Furthermore, the updated standard prescribes fewer elevation levels and employs lighter backgrounds to distinguish between them in dark mode. In my view, it would be more beneficial for Quasar to adhere to these standards rather than implementing solutions based on personal opinions.

**Comment by rstoenescu** [comment]
Shipped on `dev` in d74522273; it will be part of the next minor (v2.31).

**What changed**

The shadow color is now a pair of CSS custom properties declared on `:root`, so it can be changed from plain CSS in every flavour, the UMD and the prebuilt `quasar.css` included:

| Custom property | Default | Used |
| --- | --- | --- |
| `--q-shadow-color` | `#000` | in light mode |
| `--q-dark-shadow-color` | `#fff` | in Dark Mode and on dark component surfaces |

Reverting to the pre v2.11 look (dark shadows in Dark Mode too) is a one-liner:

```css
body.body--dark {
  --q-dark-shadow-color: #000;
}
```

`transparent` removes the shadows altogether, if you prefer the flat look. Every shadow follows: the `shadow-N` / `shadow-up-N` / `inset-shadow` helpers, QLayout's header/footer/drawer glow and the shadows of QCard, QMenu, QTable, QDate, QTime, QColor, QStepper, QUploader, QChip and QKnob. Set the properties on `:root` or on `body` (a `body.body--dark` rule as above, or `setCssVar()`); they are read at the `body` level, so an override placed deeper in the DOM has no effect.

**Why the default stays**

Light shadows on dark surfaces was a deliberate design decision in v2.11 and it has been the default for close to four years; flipping it back now would be a second visual break, this time for everyone who has built on it since. The Sass override several of you posted here (`$dark-shadow-color: #000`) keeps working unchanged, and it now has a runtime equivalent for projects without a Sass pipeline. Nothing changes visually unless you opt in.

Docs: https://quasar.dev/style/dark-mode#shadows-in-dark-mode and https://quasar.dev/style/shadows#shadow-color (live once the site deploys with v2.31).

Thanks to everyone who weighed in, and to @lianee for the CSS variables codepen that pointed the way.
